resource-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/resource/kafka/KafkaContractTest.java

```diff
@@ -237,31 +215,52 @@ void testNoSuchFileExceptionIsThrown() {
-                                        .isEqualTo(ContractTestData.REQUEST_TOKEN.getBytes()))
+                                        .isEqualTo(ContractTestData.REQUEST_TOKEN.getBytes())),
+
+                () -> assertAll("Start and End of stream markers have the correct headers",
+                        () -> assertThat(results.getFirst().headers().lastHeader(StreamMarker.HEADER).value())
+                                .isEqualTo(StreamMarker.START.toString().getBytes()),
+                        () -> assertThat(results.getLast().headers().lastHeader(StreamMarker.HEADER).value())
+                                .isEqualTo(StreamMarker.END.toString().getBytes())
+                ),
+
+                () -> assertAll("After removing the start and end of stream message",
+                        results::removeFirst,
+                        results::removeLast,
+                        () -> assertThat(results).isEmpty()
+                )
         );
 
-        // The first and last have a correct StreamMarker header
-        assertAll("StreamMarkers are correct START and END",
-                () -> assertThat(results.getFirst().headers().lastHeader(StreamMarker.HEADER).value())
-                        .isEqualTo(StreamMarker.START.toString().getBytes()),
 
-                () -> assertThat(results.getLast().headers().lastHeader(StreamMarker.HEADER).value())
-                        .isEqualTo(StreamMarker.END.toString().getBytes())
-        );
+        // When - results are pulled from the error stream
+        Probe<ConsumerRecord<String, JsonNode>> errorResultSeq = errorProbe.request(1);
+        LinkedList<ConsumerRecord<String, JsonNode>> errorResults = LongStream.range(0, 1)
+                .mapToObj(i -> errorResultSeq.expectNext(new FiniteDuration(20, TimeUnit.SECONDS)))
+                .collect(Collectors.toCollection(LinkedList::new));
 
-        // All but the first and last have the expected message
-        results.removeFirst();
-        results.removeLast();
-        assertThat(results).isEmpty();
+        // Then - the results are as expected
+        assertAll("Asserting on the error topic",
+                // One error is produced
+                () -> assertThat(errorResults)
+                        .hasSize(1),
+
+                // The error has the relevant headers, including the token
+                () -> assertThat(errorResults)
+                        .allSatisfy(result ->
+                                assertThat(result.headers().lastHeader(Token.HEADER).value())
+                                        .isEqualTo(ContractTestData.REQUEST_TOKEN.getBytes())),
+
+                // The error has a message that contains the throwable exception, and the message
+                () -> assertThat(errorResults.get(0).value().get("error").get("message").asText())
+                        .startsWith("Exception thrown while connecting to the service")
+        );
```

Comment:
This is the wrong error message and should be:
```suggestion
                () -> assertThat(errorResults.get(0).value().get("error").get("cause").get("message").asText())
                        .as("When getting the error message from the error topic")
                        .isEqualTo("Failed to walk path \\not\\a\\resource")
        );
```

Reply:
This is correct based on how the error is dealt with in the PersistenceProxy class however, I have made a change to this now so that the message from the thrown exception within the service is used.
It will now use the "Failed to walk path....." message.